aiidalab_widgets_base/structures.py

```diff
@@ -160,3 +178,38 @@ def node_class(self):
     @node_class.setter
     def node_class(self, value):
         self.data_format.value = value
+
+    @property
+    def structure_node(self):
```

Comment:
Ok, now I understand your change.

I'm still a bit unsure whether this is the right approach but I get there is value in being able to fetch both a stored and an unstored node "on demand".
That would have to be a very customized traitlet, your implementation makes more sense.